app/forms/bulk_update/bulk_add_trainees_upload_form.rb

```diff
@@ -41,6 +38,28 @@ def csv
 
   private
 
+    def create_upload
+      ActiveRecord::Base.transaction do
+        upload = BulkUpdate::TraineeUpload.create!(
+          provider: provider,
+          file: File.read(file),
```

Comment:
Before we roll this out this, would be ideal to delegate the uploaded files to S3 otherwise we would have to write migrations for this change. I think 1MB are a lot of data to `read/write/index` per row and could become a bottleneck for `Postgres` with memory consumption issues etc. Additionally, would be nice to be consistent with the rest of the `BulkUpdate` models like `Placement` and `RecomendationsUpload` which make use of it. Having said that, I don't think it is a blocker if the preference is to implement the change as part of a separate follow up PR.